### ISSUE TITLE
Check that JSON responses follow a certain format

### ISSUE THREAD

**Issue by csparpa** [sky-shiny/smolder#7]
It would be very nice to check that JSON response bodies obey a certain JSON format spec  - which  can be specified using some kind of definition language / JSON template.

Eg: I need to check that the responses contain a list of JSON objects, each one obeying the format:

```
{
  "field1" : <string>,
  "field2":  <integer>,
  "field3": <list of strings>
}
```

**Comment by sky-shiny** [comment]
Is there a particular json validation framework that people want?  Does it need to be pluggable?

**Comment by lotia** [comment]
jsonschema https://python-jsonschema.readthedocs.org/en/latest/ may be a good fit for this feature request. I may be able to provide a sample implementation is there's interest.

**Comment by sky-shiny** [comment]
Implemented in v0.2 branch as a plugin.  Branch will be master shortly.
